### PR TITLE
[LOOP-1365] Add basal schedule syncing to PumpManager

### PR DESCRIPTION
**Pull request by mpangburn**
Implementation of https://tidepool.atlassian.net/browse/LOOP-1365

See also:
- DashPumpManager implementation: https://github.com/tidepool-org/DashKit/pull/34
- MinimedPumpManager and OmnipodPumpManager implementations: https://github.com/tidepool-org/rileylink_ios/pull/14

### CODE REVIEW

File: MockKit/MockPumpManager.swift [hunk 1]
```diff
@@ -412,6 +412,12 @@ public final class MockPumpManager: TestingPumpManager {
     }
     
     public func setMaximumTempBasalRate(_ rate: Double) { }
+
+    public func syncBasalRateSchedule(items scheduleItems: [RepeatingScheduleValue<Double>], completion: @escaping (Result<BasalRateSchedule, Error>) -> Void) {
+        DispatchQueue.global().asyncAfter(deadline: .now() + .milliseconds(500)) {
```
Comment (mpangburn) [inline MockKit/MockPumpManager.swift:417]: (The delay here is to make UI loading states easy to demonstrate)

File: LoopKit/DeviceManager/PumpManager.swift [hunk 2]
```diff
@@ -169,6 +169,17 @@ public protocol PumpManager: DeviceManager {
...
+    /// Sync the schedule of basal rates to the pump, annotating the result with the proper time zone.
+    ///
+    /// - Precondition:
+    ///   - `scheduleItems` must not be empty.
+    ///
+    /// - Parameters:
+    ///   - scheduleItems: The items comprising the basal rate schedule
+    ///   - completion: A closure called after the command is complete
+    ///   - result: A BasalRateSchedule or an error describing why the command failed
+    func syncBasalRateSchedule(items scheduleItems: [RepeatingScheduleValue<Double>], completion: @escaping (_ result: Result<BasalRateSchedule, Error>) -> Void)
```
Comment (darinkrauss) [inline LoopKit/DeviceManager/PumpManager.swift:182]: [nit] Does this assume that the `scheduleItems` will always be consumed in the pump's time zone? It's not clear to me whether that or there would be some translation from phone's current time zone to pump's. Might want to add a comment one way or the other. Or perhaps that is already a basic assumed when communicating with the pump. If so, then it's fine.

**Review by darinkrauss — APPROVED** [review]
LGTM! 🐶

File: LoopKit/DeviceManager/PumpManager.swift (hunk 2 above)
Comment (ps2) [reply LoopKit/DeviceManager/PumpManager.swift:182]: ~~Current pump managers do set the new basal schedule using the phone's current timezone offset (`TimeZone.currentFixed`), and then send a `PumpManagerStatus` update to the `PumpManagerDelegate`, which Loop then uses to record the timezone change. This is not strictly required by the protocol, but it is a standard side effect.~~ I think it's fine to keep this method clear of TimeZone specification, as the PumpManager does expose the TimeZone, and reports any changes, and a schedule should always be interpreted with respect to the PumpManager's timezone.

**Review by ps2 — APPROVED** [review]
LGTM!

File: LoopKit/DeviceManager/PumpManager.swift (hunk 2 above)
Comment (ps2) [reply LoopKit/DeviceManager/PumpManager.swift:182]: Sorry, it's *not* a standard side effect. Timezone is only changed during a manual `setTime` command. This doesn't change the interpretation of a schedule being interpreted by the Pump's timezone. I'm just clarifying that it looks like the PumpManagers currently do not update their timezone in response to a basal schedule sync.

**Pull request merged**